### PR TITLE
Getting Started Guide

### PR DESCRIPTION
**Pull request by DukeDavis12**
Includes documentation for
  - Enabling ServiceInfo.
  - Getting started Guide ( for PRI & Client-SDK).
  - Keystore Management.

Signed-off-by: Davis Benny <davis.benny@intel.com>

### CODE REVIEW

File: docs/implementation-references/getting-started-guide.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+
+4. T01 and T02
+   ```
+   $ cd <client-sdk-src>
+   $ ./build/linux-client
+   ```
+
+Wait for T0 protocol completed message
+
```
Comment (DukeDavis12) [inline docs/implementation-references/getting-started-guide.md:161]: NOTES missing

File: docs/implementation-references/getting-started-guide.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+3. Voucher Extension for Client-SDK device
+   ```
+   $ curl -D - --digest -u apiUser:MfgApiPass123 -XGET http://localhost:8039/api/v1/vouchers/abcdef -o voucher
+
+   $ curl -D - --digest -u apiUser:OwnerApiPass123 --header "Content-Type: application/cbor" --data-binary @voucher http://localhost:8042/api/v1/owner/vouchers/
+   ```
+
+Wait for T00 finished for <guid> message on the Owner console.
+
```
Comment (DukeDavis12) [inline docs/implementation-references/getting-started-guide.md:175]: NOTES missing

File: docs/implementation-references/getting-started-guide.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    ```
+    $ curl -D - --digest -u apiUser:MfgApiPass123 -XGET http://localhost:8039/api/v1/vouchers/0 -o voucher
+
+    $ curl -D - --digest -u apiUser:OwnerApiPass123 --header "Content-Type: application/cbor" --data-binary @voucher http://localhost:8042/api/v1/owner/vouchers/
+    ```
+
+Wait for T00 finished for <guid> message in the Owner console.
+
+!!! NOTE
+- Keystore Management needs to be taken care, if rv and owner is not running on the same machine [REFER](#keystore-management).
```
Comment (trbehera) [inline docs/implementation-references/getting-started-guide.md]: Use PRI Rendezvous server / PRI Owner server  in place of rv / owner.

File: docs/implementation-references/getting-started-guide.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+- **You can enable ServiceInfo at this stage.** [REFER](#enabling-serviceinfo-transfer)
+
+4. T01 and T02
+   ```
+   $ cd <fdo-pri-src>/component-samples/demo/device
+   $ java -jar device.jar
+   ```
+
+Wait for T0 protocol completed message and Device is Onboarded Successfully.
```
Comment (trbehera) [inline docs/implementation-references/getting-started-guide.md]: Put a snippet of the message to expect.

File: docs/implementation-references/getting-started-guide.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+# Running E2E for Client-SDK device
+
+## Building Client-SDK Source
+
+!!! NOTE For the instructions in this document, `<client-sdk-src>` refers to the path of the FDO client-sdk folder 'client-sdk-fidoiot'.
```
Comment (trbehera) [inline docs/implementation-references/getting-started-guide.md]: Suggested rephrase:
<client-sdk-src> refers to the path of the FDO Client-SDK source folder 'client-sdk-fidoiot'.

File: docs/implementation-references/getting-started-guide.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+FDO Client-SDK source can be build by:
+
+1. Install Dependencies. [REFER](https://github.com/secure-device-onboard/client-sdk-fidoiot/blob/master/docs/linux.md)
```
Comment (trbehera) [inline docs/implementation-references/getting-started-guide.md]: All the 'REFER' statements, consider making them grammatically complete statements. 

For example,

Follow instructions in the [documentation](https://github.com/secure-device-onboard/client-sdk-fidoiot/blob/master/docs/linux.md) to install dependencies.

File: docs/implementation-references/getting-started-guide.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+
+2. Execute build.sh script
+
+   ```
+   $ cd <client-sdk-src>
+   $ ./build.sh
+   ```
+
+The build script generates artifacts and stores it in `./build/` directory.
```
Comment (trbehera) [inline docs/implementation-references/getting-started-guide.md]: stores it ==> stores them

File: docs/implementation-references/getting-started-guide.md [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+    cksum_rx=$(cksum $filename | cut -d ' ' -f 1)
+    if [ $cksum_tx -eq $cksum_rx  ]; then
+      echo "Device onboarded successfully."
+      echo "Device onboarded successfully." > result.txt
+    else
+      echo "ServiceInfo file transmission failed."
+      echo "ServiceInfo file transmission failed." > result.txt
+    fi
+    ```
+This script downloads the SECURITY.MD file and checks the integrity of file against the pre-computed checksum value.
```
Comment (trbehera) [inline docs/implementation-references/getting-started-guide.md]: SECURITY.md

**Review by v-vasavi — COMMENTED** [review]
A few of my comments 
Line 20 - - FDO PRI source can be build in two ways:  - Should it be built?
Line 39 - - Steps to run and build docker-compose up – Is the up needed here 
Line 67 - -To0scheduling interval can be adjusted in the component-sample/demo/owner/owner.env  -- Is this a variable name or do we need a space here?
Line 93 - -Wait for T00 finished for <guid> message in the Owner console.  – Should the message be in “ “ ?

File: docs/implementation-references/getting-started-guide.md (hunk 4 above)
Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md]: Done.

File: docs/implementation-references/getting-started-guide.md (hunk 2 above)
Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md:175]: Done.

File: docs/implementation-references/getting-started-guide.md (hunk 6 above)
Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md]: Done.

File: docs/implementation-references/getting-started-guide.md (hunk 7 above)
Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md]: Done.

File: docs/implementation-references/getting-started-guide.md (hunk 8 above)
Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md]: Done.

File: docs/implementation-references/getting-started-guide.md [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+- [Running E2E for PRI device](#running-e2e-for-pri-device)
+- [Building Client-SDK Source](#building-client-sdk-source)
+- [Running E2E for Client-SDK device](#running-e2e-for-client-sdk-device)
+- [Enabling ServiceInfo](#enabling-serviceinfo-transfer)
+- [Keystore Management](#keystore-management)
+
+## Building FDO PRI Source
+
+!!! NOTE
+    For the instructions in this document, `<fdo-pri-src>` refers to the path of the FDO PRI folder 'pri-fidoiot'.
```
Comment (trbehera) [inline docs/implementation-references/getting-started-guide.md:20]: @v-Vasavi  FDO PRI source can be build in two ways: - Should it be built?

Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md:20]: Yes. Checked in Grammarly.

**Review by v-vasavi — DISMISSED** [review]

File: docs/implementation-references/getting-started-guide.md [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+```
+$ cd <fdo-pri-src>
+$ mvn clean install
+```
+
+2.&nbsp;Using docker container to build FDO PRI source
+
+```
+$ cd <fdo-pri-src>/build
+$ sudo docker-compose up --build
```
Comment (KiranSukhavasi) [inline docs/implementation-references/getting-started-guide.md:35]: do we need sudo or is it optional?

File: docs/implementation-references/getting-started-guide.md [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+    $ curl -D - --digest -u apiUser:MfgApiPass123 -XGET http://localhost:8039/api/v1/vouchers/0 -o voucher
+
+    $ curl -D - --digest -u apiUser:OwnerApiPass123 --header "Content-Type: application/cbor" --data-binary @voucher http://localhost:8042/api/v1/owner/vouchers/
+
+Wait for T00 finished for <guid> message in the Owner console.
+
+Expect the following message on successful TO0 completion.
+
+```
+To0 Response Wait for <guid> : 3600
```
Comment (KiranSukhavasi) [inline docs/implementation-references/getting-started-guide.md]: TO0 spell check

File: docs/implementation-references/getting-started-guide.md [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+TO0 Client finished for GUID <guid>
+```
+
+!!! NOTE
+    - [Keystore Management](#keystore-management) needs to be taken care, if PRI Rendezvous server and PRI Owner server is not running on the same machine.
```
Comment (KiranSukhavasi) [inline docs/implementation-references/getting-started-guide.md:114]: service instead server for pri owner and pri rv. please check once.

File: docs/implementation-references/getting-started-guide.md [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+    - **You can enable ServiceInfo at this stage.** [Follow the instructions](#enabling-serviceinfo-transfer) to enable ServiceInfo.
+
+4.&nbsp;T01 and T02
```
Comment (KiranSukhavasi) [inline docs/implementation-references/getting-started-guide.md]: TO1 and TO2 spell check

File: docs/implementation-references/getting-started-guide.md [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+   ```
+   $ cd <fdo-pri-src>/component-samples/demo/device
+   $ java -jar device.jar
+   ```
+
+Wait for T0 protocol completed message and Device is Onboarded Successfully.
```
Comment (KiranSukhavasi) [inline docs/implementation-references/getting-started-guide.md]: TO2 spell check

File: docs/implementation-references/getting-started-guide.md [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+
+3.&nbsp;Voucher Extension for Client-SDK device
+
+```
+$ curl -D - --digest -u apiUser:MfgApiPass123 -XGET http://localhost:8039/api/v1/vouchers/abcdef -o voucher
+
+$ curl -D - --digest -u apiUser:OwnerApiPass123 --header "Content-Type: application/cbor" --data-binary @voucher http://localhost:8042/api/v1/owner/vouchers/
+```
+
+Wait for T00 finished for <guid> message on the Owner console.
```
Comment (KiranSukhavasi) [inline docs/implementation-references/getting-started-guide.md]: TO0 spell check

File: docs/implementation-references/getting-started-guide.md [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+
+Expect the following message on successful TO0 completion.
+
+```
+To0 Response Wait for <guid> : 3600
+TO0 Client finished for GUID <guid>
+```
+
+!!! NOTE
+    - [Keystore Management](#keystore-management) needs to be taken care, if PRI Rendezvous server and PRI Owner server is not running on the same machine.
```
Comment (KiranSukhavasi) [inline docs/implementation-references/getting-started-guide.md:186]: Service instead Server*

File: docs/implementation-references/getting-started-guide.md [hunk 17]
```diff
@@ -0,0 +1,298 @@
...
+    - **You can enable ServiceInfo at this stage.** [Follow the instructions](#enabling-serviceinfo-transfer) to enable ServiceInfo.
+
+4.&nbsp;T01 and T02
+
+```
+$ cd <client-sdk-src>
+$ ./build/linux-client
+```
+
+Wait for T0 protocol completed message
```
Comment (KiranSukhavasi) [inline docs/implementation-references/getting-started-guide.md]: TO0 spell check

File: docs/implementation-references/getting-started-guide.md (hunk 10 above)
Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md:35]: Better to add sudo here..As for most users, docker-compose might not be added to the root group.

File: docs/implementation-references/getting-started-guide.md (hunk 11 above)
Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md]: Done.

File: docs/implementation-references/getting-started-guide.md (hunk 12 above)
Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md:114]: Sticking to `server`, as we have already used `server` naming in `Starting the FDO PRI <component>  Server`

File: docs/implementation-references/getting-started-guide.md (hunk 13 above)
Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md]: Done.

File: docs/implementation-references/getting-started-guide.md (hunk 14 above)
Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md]: Done

File: docs/implementation-references/getting-started-guide.md (hunk 15 above)
Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md]: Done.

File: docs/implementation-references/getting-started-guide.md (hunk 17 above)
Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md]: Done.

File: docs/implementation-references/getting-started-guide.md (hunk 16 above)
Comment (DukeDavis12) [reply docs/implementation-references/getting-started-guide.md:186]: Sticking to Server.

**Review by KiranSukhavasi — APPROVED** [review]

**Review by trbehera — APPROVED** [review]

**Pull request merged**